pkg/ctl/create/cluster.go

```diff
@@ -387,7 +305,7 @@ func doCreateCluster(p *api.ProviderConfig, cfg *api.ClusterConfig, nameArg stri
 				return err
 			}
 
-			if err := checkEachNodeGroup(cfg, canUseForPrivateNodeGroups); err != nil {
+			if err := CheckEachNodeGroup(ngFilter, cfg, canUseForPrivateNodeGroups); err != nil {
```

Comment:
Unrelated: I would move these checks like `canUseForPrivateNodeGroups` to actual "private" functions at the end of these file so this function is shorted and contains the high level stuff. Then details like `canUseForPrivateNodeGroups` can be somewhere else and looked up later if needed. That way the function stays more relevant for the reader and less cluttered.  What do you think about that?

Reply:
Indeed, most of it needs refactoring! There is a lot of stuff that could be done in a better way. This PR only moves some stuff over to another file, but it would be great if we could refine this mess into a few packages. A stretch goal would be to turn it all into something that can be consumed as a library more easily, and something we can unit-test too!